src/utils/hooks/useClickOutside.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+
+/**
+ * useClickOutside - Handle click outside a referenced element (e.g. Popover)
+ */
+type UseClickOutside = (ref: { current: HTMLElement | null | undefined }, handler: (ev) => void) => void;
+
+const useClickOutside: UseClickOutside = (ref, handler) => {
+    useEffect(() => {
+        const handleClose = event => {
```

Comment:
Can this `event` be typed?

Reply:
Unfortunately, I didn't find a way how to type this so that the build wouldn't be failing... 

See the **handleClose** method in which we pass **event** down to **handler** method from second argument - that handler method in Popover component is **handleOut** method that in scope of component deals both with handling closing the Popover content on Keyboard event (using Enter, Space, or ESC keys) as well as with handling closing the Popover content on click outside / MouseEvent.

I spent quite some time trying to type all of those events, but I was getting lost in endless loop of Typescript errors and suggested solutions from TS logs that didn't work. (the _Type 'ABC' is not assignable to type 'DEF'_ kind of suggestions)

**KeyboardEvent**, **MouseEvent**, **SyntheticEvent**, **TouchEvent**, even using **unknown** and **never** would cause the build to fail and suggest other expected interface (that would also fail).

So, I've left either **any** or no type at all, wherever it was necessary.

I could research this more sometime close in the future, so that we can have a guideline for properly typing all of our events like these.

Reply:
Ok no worries, sometimes TS is a bit unruly